wallet/src/routes.rs

```diff
@@ -834,6 +809,105 @@ pub async fn get_last_keystore(path: &Option<PathBuf>) -> Result<Option<PathBuf>
     Ok(read_last_path(path).await?)
 }
 
+// Get the set of assets associated with the given codes.
+//
+// The caller must ensure that each asset code is known to `wallet`.
+async fn get_assets(wallet: &Wallet, codes: &[AssetCode]) -> HashMap<AssetCode, AssetInfo> {
+    iter(codes)
+        .then(|code| async move { (*code, wallet.asset(*code).await.unwrap()) })
+        .collect()
+        .await
+}
+
+async fn get_sending_account(wallet: &Wallet, address: UserAddress) -> Account {
+    let (records, asset_codes): (Vec<_>, Vec<_>) = wallet
+        .records()
+        .await
+        .filter_map(|record| {
+            if record.ro.pub_key.address() == address.0 {
+                Some((record.clone().into(), record.ro.asset_def.code))
+            } else {
+                None
+            }
+        })
+        .unzip();
+
+    Account {
+        records,
+        // `assets` is only applicable for viewing and freezing accounts.
```

Comment:
Does this mean `assets` can only be viewed or frozen, but not spent? Then why does it belong to the sending account?

Reply:
Oh I should have removed this